velox/exec/TraceUtil.h

```diff
@@ -88,6 +88,14 @@ std::string getNodeName(
     const std::shared_ptr<filesystems::FileSystem>& fs,
     memory::MemoryPool* pool);
 
+/// Extracts the hive connector from the trace metadata file, return empty if
+/// it does not exist.
+std::string getHiveConnectorId(
```

Comment:
It seems that we parse task metadata in multiple utilities. Can we consolidate this to parse once and return all the required fields? Thanks!